components/eam/src/physics/cam/phys_control.F90

```diff
@@ -262,6 +262,14 @@ subroutine phys_ctl_readnl(nlfile)
       close(unitn)
       call freeunit(unitn)
    end if
+   if (history_chemdyg_summary) then
+        if (history_gaschmbudget_2D .or. history_gaschmbudget_2D_levels) then
+            history_chemdyg_summary = .false.
+            history_gaschmbudget_2D = .true.
+            history_gaschmbudget_2D_levels = .true.
```

Comment:
Do you intend to have both 2D flags true when either 2D flags is true? Or these two 2D flags have to be both true or false? Are they connected?

Reply:
Yes, but 2D and 2D_levels can run independtly. The flag history_gaschembudget_2D_levels controls the output xxx_trop, so it is useful to be on.

Reply:
What plots the xxx_trop variables are used for? It seems that the summary flag already has the STE ozone flux in [this example](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.lee1061/20230425.amip.e99b02.FourthSmoke_chem_corrected.chrysalis/e3sm_chem_diags/plots/).

Reply:
Sorry for making it confusing. 
When I designed ```history_chemdyg_summary```, the intention is to replace ```history_gaschmbudget_2D``` and ```history_gaschmbudget_2D_levels```, so in the code, the summary flag conflicts with the other two.  ```history_chemdyg_summary``` generates variables(e.g., O3_2DTDx, xxx_trop, etc.) were controlled by both '2D' and '2D_levels flags'. This modification is assuming users accidentally turn on  ```history_chemdyg_summary``` and one of ```history_gaschmbudget_2D``` and ```history_gaschmbudget_2D_levels```. In this case, we force turn off ```history_chemdyg_summary``` but turn on both ```history_gaschmbudget_2D``` and ```history_gaschmbudget_2D_levels```. Thus, we can get the most outputs for ChemDyg.